patch/driver-hwmon-pmbus-dps200.patch

```diff
@@ -0,0 +1,176 @@
+Add PMBUS driver for DPS200 Power Supply
+
+From: Saranpong Chgobtrong <schobtr@celestica.com>
+
+Signed-off-by: Saranpong Chobtrong <schobtr@celestica.com>
+---
+ drivers/hwmon/pmbus/Kconfig  |   10 +++
+ drivers/hwmon/pmbus/Makefile |    1 
+ drivers/hwmon/pmbus/dps200.c |  130 ++++++++++++++++++++++++++++++++++++++++++
+ 3 files changed, 141 insertions(+)
+ create mode 100644 drivers/hwmon/pmbus/dps200.c
+
+diff --git a/drivers/hwmon/pmbus/Kconfig b/drivers/hwmon/pmbus/Kconfig
+index 505ca2be0..fb4b0bbae 100644
+--- a/drivers/hwmon/pmbus/Kconfig
++++ b/drivers/hwmon/pmbus/Kconfig
+@@ -218,4 +218,14 @@ config SENSORS_DPS1900
+ 	  This driver can also be built as a module. If so, the module will
+ 	  be called dps1900.
+ 
++config SENSORS_DPS200
++	tristate "Delta DPS200"
++	default n
++	help
++	  If you say yes here you get hardware monitoring support for Delta
++	  DPS200.
++
++	  This driver can also be built as a module. If so, the module will
++	  be called dps200.
++
+ endif # PMBUS
+diff --git a/drivers/hwmon/pmbus/Makefile b/drivers/hwmon/pmbus/Makefile
+index c76820a4d..f072288f6 100644
+--- a/drivers/hwmon/pmbus/Makefile
++++ b/drivers/hwmon/pmbus/Makefile
+@@ -23,3 +23,4 @@ obj-$(CONFIG_SENSORS_UCD9000)	+= ucd9000.o
+ obj-$(CONFIG_SENSORS_UCD9200)	+= ucd9200.o
+ obj-$(CONFIG_SENSORS_ZL6100)	+= zl6100.o
+ obj-$(CONFIG_SENSORS_DPS1900)	+= dps1900.o
++obj-$(CONFIG_SENSORS_DPS200)	+= dps200.o
+diff --git a/drivers/hwmon/pmbus/dps200.c b/drivers/hwmon/pmbus/dps200.c
+new file mode 100644
+index 000000000..578cc1b2c
+--- /dev/null
++++ b/drivers/hwmon/pmbus/dps200.c
+@@ -0,0 +1,130 @@
++// SPDX-License-Identifier: GPL-2.0-or-later
++/*
++ * Hardware monitoring driver for Delta DPS200
++ *
++ * (C) Copyright 2019, Celestica Inc.
++ * Author: Pradchaya Phucharoen <pphuchar@celestica.com>
++ *
++ * This program is free software; you can redistribute it and/or modify
++ * it under the terms of the GNU General Public License as published by
++ * the Free Software Foundation; either version 2 of the License, or
++ * (at your option) any later version.
++ *
++ * This program is distributed in the hope that it will be useful,
++ * but WITHOUT ANY WARRANTY; without even the implied warranty of
++ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++ * GNU General Public License for more details.
++ *
++ * You should have received a copy of the GNU General Public License
++ * along with this program; if not, write to the Free Software
++ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
```

Comment:
As SDPX header is used in the top, you do not need to add the license text.

Reply:
I have updated the license text.
And add a comment about the datasheet and described why the generic driver cannot be used.

About contacting Linux upstream.
I didn't contact the upstream. Because the generic driver has apparently been described in the source code.
If we really want to contact the upstream. Can I refer you to the question?

Reply:
Well, the rules for SONiC are to try to get it upstream. I am not getting paid for working on SONiC, so it should be your job.

Reply:
agree with paul menzel. please contact upstream